src/nft/components/sell/modal/ListingButton.tsx

```diff
@@ -0,0 +1,256 @@
+import { useOnClickOutside } from 'hooks/useOnClickOutside'
+import { Box } from 'nft/components/Box'
+import { Row } from 'nft/components/Flex'
+import { ArrowRightIcon, HazardIcon, LoadingIcon, XMarkIcon } from 'nft/components/icons'
+import { bodySmall } from 'nft/css/common.css'
+import { useNFTList, useSellAsset } from 'nft/hooks'
+import { ListingStatus, WalletAsset } from 'nft/types'
+import { pluralize } from 'nft/utils/roundAndPluralize'
+import { useEffect, useMemo, useRef, useState } from 'react'
+
+import * as styles from './ListingModal.css'
+import { getListings } from './utils'
+
+interface ListingButtonProps {
+  onClick: () => void
+  buttonText: string
+  showWarningOverride?: boolean
+}
+
+export const ListingButton = ({ onClick, buttonText, showWarningOverride = false }: ListingButtonProps) => {
+  const sellAssets = useSellAsset((state) => state.sellAssets)
+  const addMarketplaceWarning = useSellAsset((state) => state.addMarketplaceWarning)
+  const removeAllMarketplaceWarnings = useSellAsset((state) => state.removeAllMarketplaceWarnings)
+  const listingStatus = useNFTList((state) => state.listingStatus)
+  const setListingStatus = useNFTList((state) => state.setListingStatus)
+  const setListings = useNFTList((state) => state.setListings)
+  const setCollectionsRequiringApproval = useNFTList((state) => state.setCollectionsRequiringApproval)
+  const [showWarning, setShowWarning] = useState(false)
+  const [canContinue, setCanContinue] = useState(false)
+  const warningRef = useRef<HTMLDivElement>(null)
+  useOnClickOutside(warningRef, () => {
+    setShowWarning(false)
+  })
+
+  useEffect(() => {
+    const [newCollectionsToApprove, newListings] = getListings(sellAssets)
+    setListings(newListings)
+    setCollectionsRequiringApproval(newCollectionsToApprove)
+    setListingStatus(ListingStatus.DEFINED)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [sellAssets])
+
+  const [
+    noMarketplacesSelected,
+    missingExpiration,
+    invalidExpiration,
+    overMaxExpiration,
+    listingsMissingPrice,
+    listingsBelowFloor,
+    listingsAboveSellOrderFloor,
+    invalidPrices,
+  ] = useMemo(() => {
+    const noMarketplacesSelected = sellAssets.some(
+      (asset: WalletAsset) => asset.marketplaces === undefined || asset.marketplaces.length === 0
+    )
+    const missingExpiration = sellAssets.some((asset) => {
+      return asset.expirationTime != null && asset.expirationTime - Date.now() / 1000 < 60
+    })
+    const invalidExpiration = sellAssets.some((asset) => {
+      return asset.expirationTime != null && isNaN(asset.expirationTime)
+    })
+    const overMaxExpiration = sellAssets.some((asset) => {
+      return asset.expirationTime != null && asset.expirationTime - Date.now() / 1000 > 60 * 60 * 24 * 31 * 6
```

Comment:
you can actually use ms macro to convert this directly into milliseconds `ms`6 months`` should work